### ISSUE TITLE
IHtmlLocalizer not working in Core 1.0

### ISSUE THREAD

**Issue by biserkazinreich** [aspnet/Mvc#5139]
I'm succesfully using the _IViewLocalizer_, but I just can't seem to get the _IHtmlLocalizer_ working. I have created the _SharedResource_ default file `SharedResource.resx` and than another localized file `SharedResource.sl.resx` like shown on the image below.
![image](https://cloud.githubusercontent.com/assets/19290347/17643267/f83f798c-6164-11e6-878b-446864792455.png)

In my _viewImports_ I'm than doing `@inject IHtmlLocalizer<Resources.SharedResource> SharedLocalizer` and later using the `SharedLocalizer["ResourceName"]` on my views but it does not display the value from the resource file, it just displays the "ResourceName". I've also tried using the IHtmlLocalizer<Resources.SharedResource> on a controller, but I get _"No manifests exist for the current culture"_.

Here's also my _startup.cs_:
```
public class Startup
{
    public Startup(IHostingEnvironment env)
    {
        var builder = new ConfigurationBuilder()
            .SetBasePath(env.ContentRootPath)
            .AddJsonFile("appsettings.json")
            .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true);

        if (env.IsDevelopment())
        {
            LinkID=532709
            builder.AddUserSecrets();
        }

        builder.AddEnvironmentVariables();
        Configuration = builder.Build();
        AutoMapperConfig.CreateAutoMappers();
    }

    public IConfigurationRoot Configuration { get; set; }

    // This method gets called by the runtime. Use this method to add services to the container.
    public void ConfigureServices(IServiceCollection services)
    {
        services.AddDbContext<ApplicationDbContext>(options =>
            options.UseSqlServer(Configuration["Data:DefaultConnection:ConnectionString"]));

        services.AddIdentity<ApplicationUser, IdentityRole>()
            .AddEntityFrameworkStores<ApplicationDbContext>()
            .AddDefaultTokenProviders();

        services.AddMvc(o =>
        {
            o.Filters.Add(typeof(ResourceFilter));
        })
            .AddViewLocalization(LanguageViewLocationExpanderFormat.SubFolder, options => options.ResourcesPath = "Resources")
            .AddDataAnnotationsLocalization();
        services.AddScoped<LocalizationFilter>();
        services.AddLocalization(options => options.ResourcesPath = "Resources");

        services.AddTransient<IEmailSender, AuthMessageSender>();
        services.AddTransient<ISmsSender, AuthMessageSender>();
        services.AddTransient<IApplicationUserService, ApplicationUserService>();
        services.AddTransient<IUserService, UserService>();

        var supportedCultures = new List<CultureInfo>{
            new CultureInfo("en-US"),
            new CultureInfo("sl-SI"),
            new CultureInfo("de-DE"),
            new CultureInfo("hr-HR")
        };
        services.Configure<RequestLocalizationOptions>(options =>
        {
            options.SupportedCultures = supportedCultures;
            options.SupportedUICultures = supportedCultures;
            options.DefaultRequestCulture = new RequestCulture(new CultureInfo("en-US"), new CultureInfo("en-US"));
        });
    }

    public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
    {
        loggerFactory.AddConsole(Configuration.GetSection("Logging"));
        loggerFactory.AddDebug();

        if (env.IsDevelopment())
        {
            app.UseBrowserLink();
            app.UseDeveloperExceptionPage();
            app.UseDatabaseErrorPage();
        }
        else
        {
            app.UseExceptionHandler("/Site/Error");

            try
            {
                using (var serviceScope = app.ApplicationServices.GetRequiredService<IServiceScopeFactory>().CreateScope())
                {
                    serviceScope.ServiceProvider.GetService<ApplicationDbContext>().Database.Migrate();
                }
            }
            catch { }
        }

        app.UseSession();

        app.UseStaticFiles();

        app.UseIdentity();
           
        app.UseMvc(routes =>
        {
            routes.MapRoute(
                name: "LanguageUser",
                template: "{language=en}/{username=company}/{controller=Site}/{action=Index}/{id?}");
        });

        var supportedCultures = new List<CultureInfo>{
            new CultureInfo("en-US"),
            new CultureInfo("sl-SI"),
            new CultureInfo("de-DE"),
            new CultureInfo("hr-HR")
        };
        var requestLocalizationOptions = new RequestLocalizationOptions
        {
            SupportedCultures = supportedCultures,
            SupportedUICultures = supportedCultures,
            DefaultRequestCulture = new RequestCulture(new CultureInfo("en-US"), new CultureInfo("en-US"))

        };
        app.UseRequestLocalization(requestLocalizationOptions);

        Seed.Initialize(app.ApplicationServices);
    }

    public static void Main(string[] args)
    {
        var host = new WebHostBuilder()
            .UseKestrel()
            .UseContentRoot(Directory.GetCurrentDirectory())
            .UseIISIntegration()
            .UseStartup<Startup>()
            .Build();

        host.Run();
    }
}
```
As I said the _IViewLocalizer_ is working as expected so I have suspicions that I'm doing something wrong with the resource files. For that reason I've also tried renaming the line in _SharedResource.Designer.cs_ from

> new global::System.Resources.ResourceManager("**Portfolio.SharedResource**", typeof(SharedResource).GetTypeInfo().Assembly);

to

> new global::System.Resources.ResourceManager("**Portfolio.Resources.SharedResource**", typeof(SharedResource).GetTypeInfo().Assembly);

Any idea what I could be doing wrong? Thank you for any help.

**Comment by ryanbrandenburg** [comment]
@biserkazinreich firstly I would advise against editing the auto-generated *.Designer.cs files. Secondly, I'm unable to reproduce your problem with the given information, `@inject`'ed `IHtmlLocalizer`'s work just fine on the app I ginned up to test this. One thing I noticed is that you did `o.Filters.Add(typeof(ResourceFilter));` which seems suspicious, but I can't really say without being able to reproduce. If you upload a complete reproduction app as a GitHub repo I'll be better able to tell what it is that's going wrong here. It doesn't have to be pretty or even minimal, just so long as it displays this behavior.
